src/logic/prepare/provider/security.ts

```diff
@@ -0,0 +1,96 @@
+import type {
+  ApiKeyPlacement,
+  ApiKeySecurityScheme,
+  SecurityScheme,
+  SecurityValues,
+} from '@superfaceai/ast';
+import {
+  HttpScheme,
+  prepareSecurityValues,
+  SecurityType,
+} from '@superfaceai/ast';
+import inquirer from 'inquirer';
+
+export async function selectSecurity(
+  provider: string
+): Promise<{ value?: SecurityValues; scheme?: SecurityScheme }> {
+  const scheme = await enterSecuritySchema(provider);
+
+  if (scheme === 'none') {
+    return {};
+  }
+
+  return {
+    value: prepareSecurityValues(provider, [scheme])[0],
+    scheme,
+  };
+}
+
+async function enterSecuritySchema(
+  provider: string
+): Promise<SecurityScheme | 'none'> {
+  const schema: 'api key token' | 'bearer token' | 'basic' | 'digest' | 'none' =
+    (
+      await inquirer.prompt<{
+        schema: 'api key token' | 'bearer token' | 'basic' | 'digest' | 'none';
+      }>({
+        name: 'schema',
+        message: `Select a security schema for ${provider}:`,
```

Comment:
nit: It's `scheme`, not `schema`

```suggestion
        message: `Select a security scheme for ${provider}:`,
```

---

Maybe I like this version better:

```suggestion
        message: `Select authentication method for ${provider}:`,
```

---

Wouldn't it make more sense to select auth method for service (base URL) rather than provider?
Not sure you need to repeat the provider name in each prompt 🤔 

```suggestion
        message: `Select authentication method for ${baseUrl}:`,
```